python/tests/test_ibd.py

```diff
@@ -0,0 +1,749 @@
+                continue
+            if mrca_ibd and n.id != t.mrca(0, 1):
+                continue
+
+            current_int = t.get_interval()
+            if len(interval_list) == 0:
+                interval_list.append(current_int)
+            else:
+                prev_int = interval_list[-1]
+                if not path_ibd and prev_int[1] == current_int[0]:
+                    interval_list[-1] = (prev_int[0], current_int[1])
+                elif prev_dict is not None and subtrees_are_equal(
+                    t, prev_dict, node_id
+                ):
+                    interval_list[-1] = (prev_int[0], current_int[1])
+                else:
+                    interval_list.append(current_int)
+
+            prev_dict = t.get_parent_dict()
+
+        for interval in interval_list:
+            if min_length == 0 or interval[1] - interval[0] > min_length:
+                orig_id = node_map.index(node_id)
+                ibd_list.append(ibd.Segment(interval[0], interval[1], orig_id))
+
+    return ibd_list
+
+
+def get_ibd_all_pairs(
+    treeSequence,
+    samples=None,
+    min_length=0,
+    max_time=None,
+    path_ibd=True,
+    mrca_ibd=False,
+):
+
+    """
+    Returns all IBD segments for all pairs of nodes in a tree sequence
+    using the naive algorithm above.
+    """
+
+    ibd_dict = {}
+
+    if samples is None:
+        samples = treeSequence.samples().tolist()
+
+    pairs = itertools.combinations(samples, 2)
+    for pair in pairs:
+        ibd_list = get_ibd(
+            pair[0],
+            pair[1],
+            treeSequence,
+            min_length=min_length,
+            max_time=max_time,
+            path_ibd=path_ibd,
+            mrca_ibd=mrca_ibd,
+        )
+        if len(ibd_list) > 0:
+            ibd_dict[pair] = ibd_list
+
+    return ibd_dict
+
+
+def subtrees_are_equal(tree1, pdict0, root):
+    """
+    Checks for equality of two subtrees beneath a given root node.
+    """
+    pdict1 = tree1.get_parent_dict()
+    if root not in pdict0.values() or root not in pdict1.values():
+        return False
+    leaves1 = set(tree1.leaves(root))
+    for l in leaves1:
+        node = l
+        while node != root:
+            p1 = pdict1[node]
+            if p1 not in pdict0.values():
+                return False
+            p0 = pdict0[node]
+            if p0 != p1:
+                return False
+            node = p1
+
+    return True
+
+
+def verify_equal_ibd(treeSequence):
+    """
+    Calculates IBD segments using both the 'naive' and sophisticated algorithms,
+    verifies that the same output is produced.
+    NB: May be good to expand this in the future so that many different combos
+    of IBD options are tested simultaneously (all the MRCA and path-IBD combos),
+    for example.
+    """
+    ts = treeSequence
+    ibd0 = ibd.IbdFinder(ts, samples=ts.samples())
+    ibd0 = ibd0.find_ibd_segments()
+    ibd1 = get_ibd_all_pairs(ts, path_ibd=True, mrca_ibd=True)
```

Comment:
This would be a good place to test the `samples` argument, by also comparing output on a random subset of the samples.